sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2914,6 +2914,21 @@ class DataFrameSuite extends QueryTest
     val df2 = (1 to 10).toDF()
     assert(df2.isLocal)
   }
+
+  test("SPARK-35886: PromotePrecision should be subexpr replaced") {
```

Comment:
Can we have a test coverage in `catalyst` module?

Reply:
Ok, let me add one.